File: billion-loops/benchmark.cpp
#include <iostream>
#include <chrono>

int main()
{
    auto start = std::chrono::high_resolution_clock::now();
    long long sum = 0, product = 1;

    for (long long i = 0; i < 1'000'000'000; ++i)
    {
        if (i % 2 == 0)
        {
            sum += i;
        }
        else
        {
            product = (product * (i % 100)) % 1'000'000'007;
        }

        if (i % 5 == 0)
        {
            sum -= i / 2;
        }
    }

    auto elapsed = std::chrono::high_resolution_clock::now() - start;
    std::cout << "Sum: " << sum << ", Product: " << product
              << ", Elapsed Time: "
              << std::chrono::duration_cast<std::chrono::milliseconds>(elapsed).count()
              << " seconds\n";

    return 0;
}

File: billion-loops/benchmark.go
package main

import (
	"fmt"
	"time"
)

func main() {
	start := time.Now()
	var sum int64 = 0
	var product int64 = 1
	const limit int64 = 1_000_000_000

	for i := int64(0); i < limit; i++ {
		if i%2 == 0 {
			sum += i
		} else {
			product = (product * (i % 100)) % 1_000_000_007
		}

		if i%5 == 0 {
			sum -= i / 2
		}
	}

	elapsed := time.Since(start)
	fmt.Printf("Final Sum: %d\n", sum)
	fmt.Printf("Final Product: %d\n", product)
	fmt.Printf("Elapsed Time: %v\n", elapsed)
}
